src/main/kotlin/org/jetbrains/exposed/dao/UUIDEntity.kt

```diff
@@ -0,0 +1,15 @@
+package org.jetbrains.exposed.dao
+
+import org.jetbrains.exposed.sql.Column
+import java.util.UUID
+
+open class UUIDTable(name: String = "", columName: String = "id") : IdTable<String>(name) {
+  override val id: Column<EntityID<String>> = varchar(columName, 36)
```

Comment:
Yes, I was using it against MySql and Maria DB. I agree that the underlying issue should be fixed, the other thought I had was allowing for an injected UUID generator, that would allow for different generators to be used depending on the business requirement.